### PR TITLE
Add to 1Password button to card page

### PR DESCRIPTION
**Pull request by malted**

### CODE REVIEW

File: app/javascript/controllers/1pass_controller.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+          autocomplete: 'postal-code',
+          value: dataset.ccAddressPostalCode,
+        },
+        {
+          autocomplete: 'country',
+          value: dataset.ccAddressCountry,
+        },
+      ],
+      notes:
+        'Plain text. You can use [Markdown](https://support.1password.com/markdown/) too.',
```
Comment (garyhtou) [inline app/javascript/controllers/1pass_controller.js]:
```suggestion
        `HCB card for ${dataset.organizationName}`,
```

File: app/javascript/controllers/1pass_controller.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    }
+    const encodedSaveRequest = encodeOPSaveRequest(saveRequest)
+    console.info(encodedSaveRequest)
```
Comment (garyhtou) [inline app/javascript/controllers/1pass_controller.js]:
```suggestion
```

File: app/views/stripe_cards/_actions.html.erb [hunk 3]
```diff
@@ -1,10 +1,32 @@
...
+      data-cc-name="<%= stripe_card.user.name %>"
+      data-cc-number="<%= stripe_card.full_card_number %>"
+      data-cc-exp="<%= stripe_card.stripe_exp_year %><%= stripe_card.stripe_exp_month.to_s.rjust(2, "0") %>"
+      data-cc-csc="<%= stripe_card.cvc %>"
+      data-cc-type="<%= stripe_card.stripe_brand %>"
+      data-cc-address-street-address="<%= stripe_card.stripe_shipping_address_line1 %>"
+      data-cc-address-city="<%= stripe_card.stripe_shipping_address_city %>"
+      data-cc-address-state="<%= stripe_card.stripe_shipping_address_state %>"
+      data-cc-address-postal-code="<%= stripe_card.stripe_shipping_address_postal_code %>"
+      data-cc-address-country="<%= stripe_card.stripe_shipping_address_country %>">
```
Comment (garyhtou) [inline app/views/stripe_cards/_actions.html.erb]:
```suggestion
      data-organization-name="<%= stripe_card.event.name %>">
```

File: package.json [hunk 4]
```diff
@@ -52,6 +52,10 @@
     "webpack-cli": "^4.10.0"
   },
   "devDependencies": {
+    "@1password/save-button": "^1.3.0",
```
Comment (garyhtou) [inline package.json]: I think this might need to be an actual dependency

File: package.json [hunk 5]
```diff
@@ -52,6 +52,10 @@
...
+    "@babel/core": "^7.18.6",
+    "@babel/preset-env": "^7.18.6",
+    "@babel/preset-react": "^7.18.6",
```
Comment (garyhtou) [inline package.json]: not sure if this is needed

File: app/views/stripe_cards/_actions.html.erb [hunk 6]
```diff
@@ -1,10 +1,32 @@
...
-<div class="mt3 flex justify-center g1">
+<div class="mt3 flex justify-center flex-wrap g1">
   <% if !stripe_card.activated? %>
     <%= render "stripe_cards/actions/activate", stripe_card: %>
   <% else %>
     <%= render "stripe_cards/actions/freeze", stripe_card: %>
     <%= render "stripe_cards/actions/rename", stripe_card: %>
   <% end %>
+
+  <% if Flipper.enabled?(:onepass_cc_save_2024_04_29, current_user) && params["show_details"] == "true" %>
```
Comment (garyhtou) [inline app/views/stripe_cards/_actions.html.erb]:
```suggestion
  <% if stripe_card.virtual? && Flipper.enabled?(:onepass_cc_save_2024_04_29, current_user) && params["show_details"] == "true" %>
```

File: app/views/stripe_cards/_actions.html.erb [hunk 7]
```diff
@@ -1,10 +1,32 @@
...
+    <onepassword-save-button
+      data-controller="1pass"
+      data-action="click->1pass#click"
+      data-onepassword-type="credit-card"
+      lang="en"
+      class="white"
+      padding="compact"
+      data-cc-title="<%= stripe_card.name %>"
+      data-cc-name="<%= stripe_card.user.name %>"
```
Comment (garyhtou) [inline app/views/stripe_cards/_actions.html.erb]:
```suggestion
      data-cc-name="<%= stripe_card.stripe_name %>"
```

File: app/views/stripe_cards/_actions.html.erb [hunk 8]
```diff
@@ -1,10 +1,32 @@
...
+
+  <% if Flipper.enabled?(:onepass_cc_save_2024_04_29, current_user) && params["show_details"] == "true" %>
+    <onepassword-save-button
+      data-controller="1pass"
+      data-action="click->1pass#click"
+      data-onepassword-type="credit-card"
+      lang="en"
+      class="white"
+      padding="compact"
+      data-cc-title="<%= stripe_card.name %>"
```
Comment (garyhtou) [inline app/views/stripe_cards/_actions.html.erb]:
```suggestion
      data-cc-title="<%= stripe_card.name.presence || "HCB card for #{stripe_card.event.name}" %>"
```

File: app/views/stripe_cards/_actions.html.erb (hunk 3 above)
Comment (garyhtou) [inline app/views/stripe_cards/_actions.html.erb]:
```suggestion
      data-cc-address-street-address="<%= stripe_card.stripe_cardholder.stripe_billing_address_line1 %>"
      data-cc-address-city="<%= stripe_card.stripe_cardholder.stripe_billing_address_city %>"
      data-cc-address-state="<%= stripe_card.stripe_cardholder.stripe_billing_address_state %>"
      data-cc-address-postal-code="<%= stripe_card.stripe_cardholder.stripe_billing_address_postal_code %>"
      data-cc-address-country="<%= stripe_card.stripe_cardholder.stripe_billing_address_country %>">
```

**Review by sampoder — APPROVED** [review]
We're going to test this. Still undecided how we roll it out

**Pull request merged**